src/Application.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\TokenSniffer;
+
+use Exception;
+use Symfony\Component\Process\Process;
+
+class Application
+{
+    private const PARAM_EXCLUDE_FILE = 'exclude-file';
+    private const PARAM_EXCLUDE = 'exclude';
+    private const PATTERNS = [
+        'storage-token' => '[0-9]+-[0-9]+-[0-9A-Za-z]{40}+',
+        'manage-token' => '[0-9]+-[0-9A-Za-z]{40}+',
+        'aws-access-key' => '(?<![A-Z0-9])[A-Z0-9]{20}(?![A-Z0-9])',
+    ];
+
+    /** @var string */
+    private $path;
+
+    /** @var array */
+    private $excludedDirs;
+
+    /** @var array */
+    private $excludedFiles;
+
+    public function __construct()
+    {
+        $input = $this->parseInput();
+        $path = $input[0];
+        $realpath = realpath($path);
+        if (!$realpath) {
+            throw new Exception(sprintf('Path "%s" parsed as "%s" does not exist', $path, $realpath));
+        }
+        $this->path = $realpath . '/';
+        $this->excludedDirs = $input[self::PARAM_EXCLUDE] ?? [];
+        $this->excludedDirs[] = '.idea';
+        $this->excludedDirs[] = '.git';
+        $this->excludedDirs = array_unique($this->excludedDirs);
+        $this->excludedFiles = $input[self::PARAM_EXCLUDE_FILE] ?? [];
+    }
+
+    public function run(): void
+    {
+        $patterns = array_map(function (string $pattern): string {
+            return '(' . $pattern . ')';
+        }, self::PATTERNS);
+        $regexp = implode('|', $patterns);
+
+        $excludedDirsParams = [];
+        if (count($this->excludedDirs)) {
+            foreach ($this->excludedDirs as $dir) {
+                $excludedDirsParams[] = '-path';
+                $excludedDirsParams[] = $this->path . $dir;
+                $excludedDirsParams[] = '-o';
+            }
+            // last -o
+            array_pop($excludedDirsParams);
+        }
+
+        $grepCommand = [
+            'grep',
+            '-rP',
+            '--line-number',
+            $regexp,
+            '{}',
+            '+',
+        ];
+
+        $cmd = array_merge([
```

Comment:
nebylo by lepší místo monstrózního commandu to rozdělit na dva? třeba pomocí Symfony Finder najít soubory (to se nemusí testovat a nebo se to dá testovat odděleně) a pak teprv předhodit `grep`, kterej by se taky testoval samostatně? míň složitá matice testů.

Reply:
V connection je tam spousta souborů a takhle přímo v bashi to trvá třeba 20s. A to jsou jen linuxové pipy. Když by se to celé tlačilo přes PHP, tak to bude několikanásobně pomalejší IMHO. 

Plus bych teda pak musel foreach ($soubory z finderu as $file) new process ($grep $file) a těch souborů bude třeba 3000 nebo 10000 takže 3000x overhead sf/process.

Reply:
A co kontrolovat jenom change sety ?

Jednou bych provedl initial check nad vším a pak by se to spouštělo jenom na change setama. Takhle jsem to udělal já v tom ^^^.

Reply:
Jo to by to vyřešilo. Zatím jsme se shodli na tom, že by se to mělo dělat na githubu. Což se uvidí, co mi na to řeknou. a podle toho bych pak tohle upravil an ty git hooky případně.